### DISCUSSION TITLE
Add Marshmallow support

### DISCUSSION THREAD

**Discussion by mvanderlee** [fastapi/fastapi#6978] [Questions] [answered]
Pydantic solves only a small part of what APIs need. 
Marshmallow and co solve everything. 

Pydantic only validates, Marshmallow serializes and deserializes.

My usecases require the api request and response to be different from the actual python object, Marshmallow allows this, but pydantic would require me to create even more classes purely for the openapi generation. 

FastApi looks awesome, and it looks like the best project for working building an api with asyncio, but this is a blocker for our team.

Edit: additional info, we currently use flask-smorest which uses marshmallow for validation, serialization and openapi generation.

**Comment by ycd** [comment]
Wouldn't be better to use this as an external package?

As far as i have seen a lot of users trying to escape from Pydantic's validators, for performance reasons. Even Pydantic is [x2.5 faster than Marshmallow](https://pydantic-docs.helpmanual.io/benchmarks/). 

Pydantic is overall great. Hence not using Pydantic for Seralization/Serizaliation allows you to use packages like `ujson`, `orjson` etc.

**Comment by mvanderlee** [comment, marked as answer]
There are other Starlette based frameworks that solve this need like [Flama](https://flama.perdy.io/) 
It probably doesn't make sense for FastAPI to add Marshmallow support.

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for reporting back and closing the issue @mvanderlee  👍

Marshmallow is great! It was the best thing I had available until I discovered Pydantic.

BTW, Pydantic is used to validate, serialize, deserialize, filter, document, etc. 🤓 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
